### DISCUSSION TITLE
Auth0 integration question

### DISCUSSION THREAD

**Discussion by Yoni-apex** [pocketbase/pocketbase#5030] [Q&A]
Hi all,
I feel like such a noob but better to ask.
I'm looking for a realtime db, and heavily considering pocketbase.
However I struggle to understand how to integrate auth.

We're currently using Auth0, and I want to use that auth against pocketbase. How can I make it so pocketbase verify and reads the bearer token I got from the third party?
Or do I need to make a whole auth flow for pocketbase as well? If yes, all I saw in the docs was against providers urls but I'm already logged in, so I'm a bit lost :/

Thanks and sorry if that's a noob questionn🥲

**Comment by ganigeorgiev** [comment]
I'm not sure I follow and I don't understand what do you mean by _"I'm already logged in"_.

I'm not familiar with Auth0 and their APIs but if it can be used as OpenID Connect (OIDC) vendor then you can connect to it from PocketBase using the generic OAuth2 OpenID Connect (OIDC) provider (usually the last ones in the PocketBase OAuth2 providers list). To configure it you'll need to provide 3 endpoints to your Auth0 service - auth url, token url and user info-retrieval url. Based on [this post](https://community.auth0.com/t/openid-discovery-url/19536) to find the above urls you can check the json returned from `https://YOUR_AUTH0_DOMAIN/.well-known/openid-configuration`.

**Reply by Yoni-apex** [reply]
Yeah sorry, kinda overcomplicated my post. However your answer was super helpful!

Thanks, I'll give it a try! 🙏🏻

**Reply by Yoni-apex** [reply]
Ok so it works with some small issues.
Basically what I meant is that my user is already logged in in the application with Auth0. So I already have a valid token etc...

When using `authWithOAuth2` in the client, it worked but asked the user to "Allow" again, and brings some popups etc... so not the best experience. I wanted to do it manually but it seems auth0 SDK only let me retrieves `accessToken`, not OIDC auth codes.

My question is can I use my current valid token for Pocketbase? I see in the JS SDK that there is the possibility to provide my own AuthStore on initialization, not sure what it should include though. And mostly also, if the user is seen for the first time, will providing my own token trigger the user creation on the PB DB?

Thanks again for all the help!

**Reply by ganigeorgiev** [reply]
> My question is can I use my current valid token for Pocketbase?

No. The Auth0 tokens are different from the tokens that PocketBase uses. Once the `authWithOAuth2` completes PocketBase no longer care about the Auth0 state.

> And mostly also, if the user is seen for the first time, will providing my own token trigger the user creation on the PB DB?

I don't understand this.

In any case, I don't think I can help further but if you need further assistance please elaborate more clearly on your use case (_eg. what is the point of using Auth0 in the first place?_) to understand the bigger picture and to avoid the needless back and forth (see also [XY problem](https://xyproblem.info/)).

**Reply by Yoni-apex** [reply]
> I don't understand this.
>
> In any case, I don't think I can help further but if you need further assistance please elaborate more clearly on your use case (eg. > what is the point of using Auth0 in the first place?) to understand the bigger picture and to avoid the needless back and forth (see > also [XY problem](https://xyproblem.info/)).

Of course.

All our system is already in place and already uses Auth0 for managing user across the system, from our BFF, to our other microservices.
However now we want to bring a real time DB and use Auth0 to auth against it to enforce the security against the record (user X can only see records attached to them).

That's the root of this topic. From their yeah I try to figure how to make them play nice together.

> No. The Auth0 tokens are different from the tokens that PocketBase uses. Once the authWithOAuth2 completes PocketBase no longer care about the Auth0 state.

I asked because since I provide my auth0 secret to PocketBase, I figured maybe it can simply verify and use it, it was just a wild guess.
